[IV-24-25] Objetivo 1

lmchaves opened the pull request:
* [x] ¿Las historias de usuario y los milestones constituyen una guía
    suficiente para poder comenzar el desarrollo de un proyecto?
Sí, cada milestone establece qué funcionalidad se va a implementar, lo que permite empezar a trabajar desde la estructura del taller hasta la implementación de la lógica de negocio.
* [x] ¿Los milestones están ordenados correctamente, de forma que el 1 o
   0 sea lo primero que hay que empezar a publicar?
Sí, el Milestone 0 está relacionado con el modelado del dominio, que es el primer paso lógico. A partir de ahí, los milestones avanzan de forma secuencial y lógica.
* [x] ¿Todos los milestones se construyen sobre el anterior, teniendo un
   orden lógico el desarrollo?
Sí.
* [x] ¿El milestone 0 tiene asignada alguna historia de usuario?
 M0 establece las bases para poder implementar las historias de usuario
* [x] ¿El milestone 0 corresponde con lo mencionado varias veces en
   clase, y en concreto con lo que se solicita en el objetivo
   siguiente?
Sí, se enfoca en modelar el dominio del problema, lo que es fundamental antes de implementar cualquier lógica de negocio.
* [x] ¿Todos los milestones describen correctamente un producto y qué
   lo hace válido, y no un  concepto vago, una funcionalidad o una tarea?
Sí, se describen claramente los criterios que hacen que el producto sea válido.
* [x] ¿Los milestones son graduales, o hay un salto grande entre ellos?
Son graduales, ya que primero se modela, luego se consulta el inventario y después se gestiona la distribución de piezas entre sedes. Esto evita saltos grandes y permite un avance progresivo.
* [x] ¿Los milestones iniciales son internos, y hay a continuación milestones o
   productos externos?
Sí, el Milestone 0 es interno (modelado del sistema), mientras que los siguientes (M1 y M2) son externos, ya que afectan la interacción con usuarios del sistema, como mecánicos y el jefe del taller.
* [x] ¿Los milestones son mínimos, es decir, incluyen un producto
   mínimamente viable para poder avanzar?
Sí.
* [x] ¿El milestone 0 es mínimo *y* viable?
Sí.
* [x] ¿Se indica claramente cómo comprobar la viabilidad del PMV de cada hito? Es
    decir, ¿se dice qué requisitos técnicos tiene que cumplir el producto de
    cada hito?
Sí, cada milestone incluye criterios de viabilidad claros.
* [x] ¿Entre dos milestones consecutivos, no hay ningún PMV posible o por el
    contrario podrían desarrollarse muchas posibles etapas internas o externas
    de un proyecto?
* [x] ¿He tenido en cuenta en las historias de usuario que se trata de un
   proyecto que se desplegará en la nube?
Sí.
* [x] ¿Están descritos todos los conceptos mencionados en las HUs con
   suficiente claridad, generalmente en documentos aparte?
* [x] ¿Tienen coherencia suficiente las historias de usuario, y en caso
   de que no lo tengan, se ha escrito un *user journey* para
   aclararlo?
Sí, hay coherencia suficiente
* [x] ¿Todas las historias de usuario representan un beneficio para el
    mismo y se relacionan con la lógica de negocio? Es decir, ¿los usuarios
    *desean* hacerlo o tú deseas que el usuario te lo pida para hacerlo?
Sí, las historias de usuario presentan beneficios claros.

lmchaves committed:
He realizado cambios en el orden de los milestones.

- En el modelado del problema trabajo con ambas HU ya que aunque dividimos el problema en dos historias de usuario (HU1 y HU2), es posible trabajar en ambas simultáneamente porque están interrelacionadas. Por ejemplo, la consulta de disponibilidad de piezas (HU1) requiere información que se gestiona en el inventario (HU2). Trabajar en conjunto permite al equipo abordar la lógica de negocio de manera más coherente y asegurar que las funcionalidades se integren adecuadamente.

- He modificado el M2 (anteriormente M1) para que el entregable quede más claro.

lmchaves committed:
Agrego los viajes de usuario

lmchaves committed:
Agrego las historias de usuario

lmchaves commented:
@JJ han pasado ya 3 días y nadie ha contestado al pull request

JJ commented on docs/README.md:
Os he dicho repetidamente que los cambios que no correspondan al objetivo no los incluyáis en el PR. Si tienes que borrar esto, y más aún si has tenido que rehacer el PR, podías haberlo borrado en main.

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Hay que realizar un análisis de las historias de usuario, en concreto de las historias de usuario [HU1](https://github.com/lmchaves/OrganizarTaller/issues/4) y [HU2](https://github.com/lmchaves/OrganizarTaller/issues/5), para identificar y definir los problemas clave relacionados con la gestión y disponibilidad de piezas en el taller. Este análisis establecerá las bases para la creación de un m ...
```
¿"la gestión y disponibilidad de piezas en el taller" está en la HU? Si está, por qué lo pones. Y si no está, ¿por qué no lo pones en la HU?

lmchaves replied:
Efectivamente esta en la HU, aqui esta cambiado: 2c88ffcc639892fea59284e06b6fc532f935551d @JJ

JJ reviewed (changes requested):
Por favor, repasa el concepto de historia de usuario tanto en el guión como [en los errores frecuentes](https://github.com/JJ/IV-/blob/master/errores/objetivo-1.md#sobre-las-historias-de-usuario)

lmchaves committed:
Update milestones.md

He borrado la parte de "la gestión y disponibilidad de piezas en el taller", ya que efectvamente esta en el HU1, pro lo que estoy dando información redundante.

JJ commented on docs/README.md:
Ya he indicado que este fichero no debe estar aquí.

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Hay que realizar un análisis de las historias de usuario, en concreto de las historias de usuario [HU1](https://github.com/lmchaves/OrganizarTaller/issues/4) y [HU2](https://github.com/lmchaves/OrganizarTaller/issues/5), para identificar y definir los problemas clave. Este análisis establecerá las bases para la creación de un modelo que represente los procesos y desafíos implicados en la gestió ...
```
"Identificar y definir los problemas clave" ¿es parte del "modelado del problema"? ¿Es un solo problema o muchos problemas? ¿Por qué se divide el problema en dos historias de usuario pero luego se pide que se "identifiquen los problemas clave" de dos HUs juntas? ¿Qué bases hay que establecer? ¿Las "bases para la creación del problema" son lo mismo que "los problemas clave"? ¿Estamos hablando de un mismo problema o problemas diferentes?

JJ reviewed (changes requested):
Creo que debéis entender que se trata de un documento técnico que sirva de base para el desarrollo de un producto por tu parte o quizás de otra persona. Un párrafo de palabrería ayuda muy poco a entender qué es lo que se tiene que hacer. Lo que se busca aquí no es que escribáis cosas para cubrir un expediente, sino que entendáis como se desarrolla un producto con mentalidad ágil.

lmchaves committed:
Update milestones.md

- La identificación de problemas clave en lugar de hacer una afirmación general, considera enumerar los problemas que pueden surgir en la gestión de inventario, distribución de piezas y atención al cliente.

- El uso de las 2 HU en el modelo del problema se debe a que aunque cada historia aborda aspectos diferentes, están interrelacionadas y afectan al mismo proceso general.
Ya qye analizarlas en conjunto nos permite identificar cómo los cambios en la gestión del inventario (HU2) impactan directamente en la consulta de disponibilidad de piezas (HU1), y viceversa.

lmchaves commented:
En realizado cambios y en el úlitmo commit he respondido a sus preguntas de la úlitma revisión. @JJ

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Hay que realizar un análisis de las historias de usuario, en concreto de las historias de usuario [HU1](https://github.com/lmchaves/OrganizarTaller/issues/4) y [HU2](https://github.com/lmchaves/OrganizarTaller/issues/5), para identificar y definir los problemas clave. para la creación de un modelo que represente los procesos y desafíos implicados en la gestión del inventario, la distribución de ...
```
¿La gestión del inventario, la distribución de piezas y la atención al cliente son parte de las historias de usuario referenciadas? Si están, ¿por qué las mencionas? Si no lo están, ¿por qué no las incluyes en las mismas?

JJ reviewed (changes requested):
Por favor, trata de entender el concepto de historia de milestone e historia de usuario. Un milestone no es un mandao, no es *haz esto*. ¿No te das cuenta que si dices "tienes que hacer la gestión de esto o de aquello" estás negando y contradiciendo todo lo demás que has dicho anteriormente y que las historias de usuario no sirven para nada? ¿No te das cuenta de que entonces las historias de usuario no sirven para nada?
Yo entiendo que estéis acostumbrados a hacer cosas que no sirven para nada y que con escribir dos o tres cosas que más o menos parezcan razonables se pase una práctica o se supere un examen. En el mundo real no es así. Tienes que escribir documentos con precisión para que una persona sepa trabajar, y no solo eso, pueda desarrollar el potencial que tiene sin necesidad de una supervisión constante, es decir, pueda responder autonónomamente a si lo que hace está bien.
¿No te das cuenta que si pones "haz esto y aquello" en realidad es más fácil que lo hagas tú? Que no se trata de que hagáis el teatrillo de los user journey, las HU y los milestone, como en tantas asignaturas. Los user journey, para empezar, se hace si son útiles (no parece que sea el caso aquí, por cierto), no *porque os he dicho que hagáis user journeys). Las historias de usuario se hacen porque son útiles para dividir el problema. Y los milestones se hacen para organizar el trabajo en productos de complejidad creciente, donde el desarrollo se pueda abordar poco a poco, con eficiencia, y con autonomía.
Ya te dije en otro comentario lo de las dos historias de usuario. Insistes en ponerlas ahí, tú verás. Por sí solo no es un problema. Pero en conjunto el problema es que no entiendes que esto sirva para nada, ni por qué se hace, ni cómo lo vas a utilizar (tú y/o otra persona) más adelante. Quizás sería conveniente que empezaras por esto último, porque si no lo otro se va a hacer, se está haciendo, muy cuesta arriba.

lmchaves committed:
Update user-journeys.md

Modifica los viajes de usuario, para que se vea el día a día de los empleados del taller y los probelmas a los que se enfrentan.

lmchaves committed:
Update user-stories.md

Modifico el HU3 añadiendo los criterios de prioridad

lmchaves committed:
Update user-stories.md

He realizado cambios en el orden y cnotenido  de los 2 primeros  HU, para aportar una mayor calridad

lmchaves committed:
Update user-stories.md

breve cambio para que se vean bien los enlaces

lmchaves committed:
Update milestones.md

He realizado cambios en el M0, explicando de manera clara y precisa el modelo del problema, he cambiado brevemente el M1 ajustandonos al HU2, y he añadido el M2, el cual antes estaba junto al M1, lo cual era érroneo y le quitaba sentido a dividir el rpoblemas en las correspondientes HU.

lmchaves committed:
Update user-stories.md

Cambio en el HU2, añadiendo ua mayor explicación para que quede claro el problema

lmchaves commented:
Gracias por tus comentarios. Quiero aclarar que no tenía la intención de hacer algo incorrecto, sino que me había enfocado en estructurar el trabajo dividiendo el problema en dos historias de usuario (HU), pensando que era la manera correcta de abordar las necesidades de los usuarios.
Estaba encabezonado con lo que hacia estaba bien, y he realizado el preoceso de poner en lugar de otra persona que coge este proyecto desde cero, y he realizado los cambios correspondientes para que quede todo quede realmente explicado y que tenga un uso útil. @JJ

JJ commented on docs/user-stories.md:
```diff
@@ -13,4 +13,4 @@ Necesito acceder a un sistema que me permita supervisar y gestionar el inventari
 ## [HU3] Jefe del Taller: Gestión de Prioridades
 
-Soy Manuel, jefe del taller, y quiero acceder a un sistema que me permita priorizar las reparaciones según la urgencia de los clientes, considerando la disponibilidad de las piezas y los tiempos de entrega estimados. Esto me permitirá asignar los recursos de manera eficiente y comunicarme con los clientes de manera más precisa sobre los plazos de entrega de sus reparaciones.
+Soy Manuel, jefe del taller, y quiero un sistema que me permita priorizar las reparaciones según la urgencia del cliente, optimizando la asignación de recursos. Para ello, el sistema debe clasificar las reparaciones de acuerdo con la urgencia indicada, la disponibilidad inmediata de las piezas, y los tiempos de entrega estimados para las piezas solicitadas cuando no estén en stock local. Esto me  ...
```
¿Te parece que esto sea un problema o una solución? ¿"El sistema debe clasificar las reparaciones" es la expresión de un problema por parte del usuario?

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,27 @@
+- __Viabilidad__:
+El hito se considerará completado cuando el algoritmo de optimización asigne correctamente las piezas solicitadas a la sede adecuada, reduciendo los costos y tiempos de manera efectiva.
+
+# [ [M2] Milestone 2: Priorización de reparaciones según urgencia](https://github.com/lmchaves/OrganizarTaller/milestone/3)
```
¿Has leído lo que dice el fichero de errores comunes con respecto a los milestones más allá del 1?

JJ commented on docs/user-journeys.md:
```diff
@@ -0,0 +1,26 @@
+# User Journeys
```
¿Te está sirviendo para algo este fichero? ¿O lo pones porque piensas que se pide?

JJ reviewed (commented):
Te iba a poner otra vez los mismos comentarios, pero no creo que sea la mejor forma de emplear el tiempo, porque ya te lo he repetido varias veces.

lmchaves committed:
Update user-stories.md

He modificado la HU2, para que sea la explicación del problema sin incluir nada de solución.

He elminado HU3 para evitar sobrecargar a quien le toque trabajar en el siguiente objetivo.

lmchaves committed:
Update milestones.md

He realizado cambios en la viabilidad del M1 para que tenga un criterio para proder aprobarlo.

Se ha eliminiado el milestone asociado a la HU3 que fue elimnada para no sobrecargar.

lmchaves commented:
He corregido tanto los milestones como las historias de usuario, y tengo una duda y es que menciono, precios sobre las piezas y costes y eso lo tengo en varioas archivos excel, y no como es información importante no se si añadirla en los user-stories, o en los user-journeys. @JJ

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,19 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Realizar un análisis de la [HU1](https://github.com/lmchaves/OrganizarTaller/issues/5) para identificar y definir los elementos clave del dominio relacionados con la supervisión y gestión del inventario de piezas en el taller. 
```
Luis, no hay que poner más detalles sobre "el dominio" que no sean los que están en la historia de usuario. Creía que eso te había quedado claro.

JJ reviewed (changes requested):
Todos los detalles del problema deben estar en la historia de usuario. Un milestone sólo debe decir cómo se empaqueta el producto y qué lo hace válido.

lmchaves committed:
Update milestones.md

He realizado cambios en el M0, para que solo sean los elemntos de la HU.

lmchaves committed:
Update milestones.md, el cambio anterior no es valido

lmchaves committed:
Update milestones.md

Cambio en el M0, para que se explique solo lo de la HU1.

lmchaves committed:
Añedo los precios de las diferentes sedes

lmchaves committed:
Update user-stories.md

Añado los archivos de  los precios

lmchaves commented:
Ya lo he arreglado @JJ

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,19 @@
+# [ [M0] Milestone 0: Modelado del problema](https://github.com/lmchaves/OrganizarTaller/milestone/4)
+- Realizar un análisis de la [HU1](https://github.com/lmchaves/OrganizarTaller/issues/5), a partir de el se procesarán los datos de la HU1 en código, para  modelizar un punto inicial sobre el que se irá trabajando.
```
Si lees esto aunque usa todas las palabras correctas, no deja de tener sentido. ¿Qué son "los datos" de una HU!? ¿Qué análisis hay que realizar? ¿Qué es modelizar "un punto inicial"?

JJ reviewed (changes requested):
Creo que sigues sin entender qué es un modelo del problema y por qué hace falta. Tampoco es "modelado". Modelado es una tarea. Modelo es un producto (si se empaqueta como tal, claro)

lmchaves committed:
Update milestones.md

He modificado el M0, para que quede calro que son los datos, entidades y las relaciones entre ellas.

En cuanto al analisis es basicamente, comprender el problema para poder ver cuales son las entidades y relaciones.

He modificado lo de "  modelizar "un punto inicial" ", para que quede claro que es un modelo inicial, en el que se irá añadiendo mas trabajo (por ejemplo el milestone1)

lmchaves committed:
Update milestones.md

CAMBIO ANTERIOR NO VALIDO

lmchaves committed:
Update milestones.md

He modificado el M0, para que quede calro que son los datos, entidades y las relaciones entre ellas.

En cuanto al analisis es basicamente, comprender el problema para poder ver cuales son las entidades y relaciones.

He modificado lo de "  modelizar "un punto inicial" ", para que quede claro que es un modelo inicial, en el que se irá añadiendo mas trabajo (por ejemplo el milestone1)

lmchaves commented:
He modificado el M0, para que quede claro @JJ

lmchaves committed:
Update user-journeys.md

Añado un ejemplo de como sería la realización de su trabajo

lmchaves committed:
Update milestones.md

Cambios en el M0, tanto el entregable sin detalles adicionales

lmchaves commented:
Ya esta corregido el M0. @JJ

lmchaves committed:
Update milestones.md

Modifico el M1, quitando sobreinformación que detallé.

lmchaves committed:
cambios anteriores no validos

lmchaves committed:
Update milestones.md

Cambios en el M1, tanto criterio como explicación.

lmchaves commented:
es el útlimo commit @JJ a81d8fc

JJ reviewed (approved):
👍
